### PR TITLE
fix: Do not create cors_configuration when variable not provided

### PR DESCRIPTION
**Pull request by durnik-ivo**
## Description
Filters out `null` values when checking for `var.cors_configuration` content and deletes default value for `var.cors_configuration.expose_headers`, which is [not required](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/apigatewayv2_api#cors_configuration).

## Motivation and Context
When `var.cors_configuration` is not passed to the module, its content is not `{}`, because a type is specified for the variable:

```
      + cors_configuration = {
          + allow_credentials = null
          + allow_headers     = null
          + allow_methods     = null
          + allow_origins     = null
          + expose_headers    = []
          + max_age           = null
        }
```

this causes condition for the dynamic block to be evaluted incorrectly:

```terraform
resource "aws_apigatewayv2_api" "this" {
  # [...]

  dynamic "cors_configuration" {
    for_each = local.is_http && length(var.cors_configuration) > 0 ? [var.cors_configuration] : []

    content {
      # [...]
    }
  }
``` 

and adds an empty cors_configuration section to the resource:

```
  # module.api_gateway_orig.aws_apigatewayv2_api.this[0] will be created
  + resource "aws_apigatewayv2_api" "this" {
      + api_endpoint                 = (known after apply)
      + api_key_selection_expression = "$request.header.x-api-key"
      + arn                          = (known after apply)
      + execution_arn                = (known after apply)
      + id                           = (known after apply)
      + name                         = "idurnik_test_module"
      + protocol_type                = "HTTP"
      + route_selection_expression   = "$request.method $request.path"
      + tags                         = {
          + "terraform-aws-modules" = "apigateway-v2"
        }
      + tags_all                     = {
          + "terraform-aws-modules" = "apigateway-v2"
        }

      + cors_configuration {} # <---- Adds empty cors section
    }
```

instead this section should not be configured:

```
  # module.api_gateway_mine.aws_apigatewayv2_api.this[0] will be created
  + resource "aws_apigatewayv2_api" "this" {
      + api_endpoint                 = (known after apply)
      + api_key_selection_expression = "$request.header.x-api-key"
      + arn                          = (known after apply)
      + execution_arn                = (known after apply)
      + id                           = (known after apply)
      + name                         = "idurnik_test_module_mine"
      + protocol_type                = "HTTP"
      + route_selection_expression   = "$request.method $request.path"
      + tags                         = {
          + "terraform-aws-modules" = "apigateway-v2"
        }
      + tags_all                     = {
          + "terraform-aws-modules" = "apigateway-v2"
        }

      # <--- no cors section
    }
```

## Breaking Changes
Yes.

## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [ ] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by yungbender** [comment]
Adding on top of this, when you propagate this
```
      + cors_configuration {}
```
Into your api gateway, it forces the gateway to start overriding your CORS headers returned from the gateway integrations, which makes this module a hassle to use.

File: main.tf [hunk 1]
```diff
@@ -16,7 +16,7 @@ resource "aws_apigatewayv2_api" "this" {
   body                         = local.is_http ? var.body : null
 
   dynamic "cors_configuration" {
-    for_each = local.is_http && length(var.cors_configuration) > 0 ? [var.cors_configuration] : []
```
Comment (bryantbiggs) [inline main.tf:19]: I don't think this is a valid change

File: variables.tf [hunk 2]
```diff
@@ -27,7 +27,7 @@ variable "cors_configuration" {
     allow_headers     = optional(list(string))
     allow_methods     = optional(list(string))
     allow_origins     = optional(list(string))
-    expose_headers    = optional(list(string), [])
```
Comment (bryantbiggs) [inline variables.tf:30]: I think this might be a valid change - will test it out

Comment (bryantbiggs) [reply variables.tf:30]: not valid - updated in https://github.com/terraform-aws-modules/terraform-aws-apigateway-v2/pull/114/commits/b7ec703569f213a510721a884430865e59550cab

**Review by bryantbiggs — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 5.1.1 :tada:

**Pull request merged**